src/main/java/ch/njol/skript/effects/EffStopSound.java

```diff
@@ -36,97 +28,113 @@
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
+import org.bukkit.Sound;
+import org.bukkit.SoundCategory;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Locale;
+import java.util.regex.Pattern;
 
 @Name("Stop Sound")
-@Description({"Stops a sound from playing to the specified players. Both Minecraft sound names and " +
+@Description({"Stops the given sound(s) from playing to the specified players. Both Minecraft sound names and " +
 		"<a href=\"https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Sound.html\">Spigot sound names</a> " +
 		"are supported. Resource pack sounds are supported too. The sound category is 'master' by default. " +
 		"A sound can't be stopped from a different category. ",
 		"",
 		"Please note that sound names can get changed in any Minecraft or Spigot version, or even removed from Minecraft itself."})
-@Examples({"stop sound \"block.chest.open\" for the player",
-		"stop playing sounds \"ambient.underwater.loop\" and \"ambient.underwater.loop.additions\" to the player"})
-@Since("2.4")
-@RequiredPlugins("Minecraft 1.10.2+, Minecraft 1.11+ (sound categories)")
+@Examples({
+	"stop sound \"block.chest.open\" for the player",
+	"stop playing sounds \"ambient.underwater.loop\" and \"ambient.underwater.loop.additions\" to the player",
+	"stop all sound for all players",
+	"stop sound in record category"
+})
+@Since("2.4, INSERT VERSION (stop all sounds)")
+@RequiredPlugins("MC 1.17.1 (stop all sounds)")
 public class EffStopSound extends Effect {
+	
+	private static final boolean STOP_ALL_SUPPORTED = Skript.methodExists(Player.class, "stopAllSounds");
+	private static final Pattern KEY_PATTERN = Pattern.compile("([a-z0-9._-]+:)?[a-z0-9/._-]+");
 
-	private static final boolean SOUND_CATEGORIES_EXIST = Skript.classExists("org.bukkit.SoundCategory");
-
+	
 	static {
-		if (Skript.methodExists(Player.class, "stopSound", String.class)) {
-			if (SOUND_CATEGORIES_EXIST) {
-				Skript.registerEffect(EffStopSound.class,
-						"stop sound[s] %strings% [(in|from) %-soundcategory%] [(from playing to|for) %players%]",
-						"stop playing sound[s] %strings% [(in|from) %-soundcategory%] [(to|for) %players%]");
-			} else {
-				Skript.registerEffect(EffStopSound.class,
-						"stop sound[s] %strings% [(in|from) %-soundcategory%] [(from playing to|for) %players%]",
-						"stop playing sound[s] %strings% [(in|from) %-soundcategory%] [(to|for) %players%]");
-			}
-		}
+		String stopPattern = STOP_ALL_SUPPORTED ? "(all:all sound[s]|sound[s] %strings%)" : "sound[s] %strings%";
+		
+		Skript.registerEffect(EffStopSound.class,
```

Comment:
Do you think this should support multiple sound categories?

Something like `stop all sounds in <category> and <category>` made sense to me.

Reply:
Only one sound can be stopped at once equalling one packet. Minecraft isn't designed to support stopping from multiple categories, besides if you don't know the category you can just omit it to stop it regardless.

The current category for stopping sounds is designed for when you do know what is to be stopped. E.g. music from the player category to note a personal jukebox and another global one from the block category. Then you stop the personal one but keep the global one active. It doesn't make much sense to me to allow for stopping multiple categories as the chance someone would need that is absolutely minimal.